docs/storage/azure-storage.md

```diff
@@ -58,188 +71,149 @@ You also need to assign the following roles to the user account you are logged i
 
 ```dotnetcli
 dotnet add package Aspire.Azure.Storage.Blobs --prerelease
 dotnet add package Aspire.Azure.Storage.Queues --prerelease
 ```
 
-In the _Program.cs_ file of the `AspireStorage` project, add calls to the <xref:Microsoft.Extensions.Hosting.AspireBlobStorageExtensions.AddAzureBlobService%2A> and <xref:Microsoft.Extensions.Hosting.AspireQueueStorageExtensions.AddAzureQueueService%2A> extension methods. Provide the name of your connection string as a parameter.
+In the _Program.cs_ file of the _AspireStorage_ project, add calls to the <xref:Microsoft.Extensions.Hosting.AspireBlobStorageExtensions.AddAzureBlobService%2A> and <xref:Microsoft.Extensions.Hosting.AspireQueueStorageExtensions.AddAzureQueueService%2A> extension methods after the creation of the `builder` but before the call to `AddServiceDefaults`. For more information, see [.NET Aspire service defaults](../service-defaults.md). Provide the name of your connection string as a parameter.
 
-```csharp
-builder.AddAzureBlobService("blobConnection");
-builder.AddAzureQueueService("queueConnection");
-```
+:::zone pivot="azurite"
+
+:::code source="snippets/tutorial/AspireStorage/AspireStorage/AspireStorage/Program.cs" highlight="3-4,8-9,27-33":::
+
+:::zone-end
+:::zone pivot="azure-portal,azure-cli"
+
+:::code source="snippets/tutorial/AspireStorage/AspireStorage/AspireStorage/Program.cs" range="1-25,34-52" highlight="3-4,8-9":::
+
+:::zone-end
 
-These methods accomplish the following tasks:
+With the additional `using` statements, these methods accomplish the following tasks:
 
 - Register a <xref:Azure.Storage.Blobs.BlobServiceClient?displayProperty=fullName> and a <xref:Azure.Storage.Queues.QueueServiceClient?displayProperty=fullName> with the DI container for connecting to Azure Storage.
 - Automatically enable corresponding health checks, logging, and telemetry for the respective services.
 
-In the _appsettings.json file of the `AspireStorage` project, add the corresponding connection information:
+:::zone pivot="azurite"
+
+When the _AspireStorage_ project starts, it will create a `fileuploads` container in Azurite Blob Storage and a `tickets` queue in Azurite Queue Storage. This is conditional when the app is running in a development environment. When the app is running in a production environment, the container and queue are assumed to have already been created.
+
+:::zone-end
+
+In the _appsettings.json_ file of the _AspireStorage_ project, add the corresponding connection information:
 
 ```json
 "ConnectionStrings": {
-  "blobConnection": "https://{account_name}.blob.core.windows.net/",
-  "queueConnection": "https://{account_name}.queue.core.windows.net/"
+  "BlobConnection": "https://{account_name}.blob.core.windows.net/",
+  "QueueConnection": "https://{account_name}.queue.core.windows.net/"
 }
```

Comment:
We had mentioned moving these to the apphost project because of the addition of withreference - did that not work out?

Reply:
Yes, I need to move the config to the app host so it's shared. Then the app host will be shown for all pivots.

Reply:
I spoke with @eerhardt about this a bit yesterday. At one point we were thinking it made sense to put this config in the app host, and have the app host be responsible for ensuring that the resources were created when using Azurite, instead of having each project ensure their own resources were created.

Let's see how this discussion turns out here: https://github.com/dotnet/docs-aspire/pull/108#discussion_r1412308197